File: docs/public/specs/esm/earthquakes-globe.js
import * as vg from "@uwdata/vgplot";

await vg.coordinator().exec([
  vg.loadExtension("spatial"),
  vg.loadParquet("earthquakes", "data/earthquakes.parquet"),
  vg.loadSpatial("land", "data/countries-110m.json", {layer: "land"})
]);

const $longitude = vg.Param.value(-180);
const $latitude = vg.Param.value(-30);
const $rotate = vg.Param.array([$longitude, $latitude]);

export default vg.vconcat(
  vg.hconcat(
    vg.slider({label: "Longitude", as: $longitude, min: -180, max: 180, step: 1}),
    vg.slider({label: "Latitude", as: $latitude, min: -90, max: 90, step: 1})
  ),
  vg.plot(
    vg.geo(
      vg.from("land"),
      {geometry: vg.geojson("geom"), fill: "currentColor", fillOpacity: 0.2}
    ),
    vg.sphere(),
    vg.dot(
      vg.from("earthquakes"),
      {
        x: "longitude",
        y: "latitude",
        r: vg.sql`POW(10, magnitude)`,
        stroke: "red",
        fill: "red",
        fillOpacity: 0.2
      }
    ),
    vg.margin(10),
    vg.style("overflow: visible;"),
    vg.projectionType("orthographic"),
    vg.projectionRotate($rotate)
  )
);

File: docs/public/specs/python/earthquakes-globe.py
import vgplot as vg

earthquakes = vg.parquet("data/earthquakes.parquet")
land = vg.spatial("data/countries-110m.json", layer="land")

longitude = vg.param(-180)
latitude = vg.param(-30)
rotate = vg.param([longitude, latitude])

view = vg.vconcat(
    vg.hconcat(
        vg.slider(label="Longitude", bind=longitude, min=-180, max=180, step=1),
        vg.slider(label="Latitude", bind=latitude, min=-90, max=90, step=1),
    ),
    vg.plot(
        vg.geo(
            land, geometry=vg.geojson("geom"), fill="currentColor", fill_opacity=0.2
        ),
        vg.sphere(),
        vg.dot(
            earthquakes,
            x="longitude",
            y="latitude",
            r=vg.sql("POW(10, magnitude)"),
            stroke="red",
            fill="red",
            fill_opacity=0.2,
        ),
        vg.margin(10),
        vg.style("overflow: visible;"),
        vg.projection_type("orthographic"),
        vg.projection_rotate(rotate),
    ),
)

